### PR TITLE
plugins.zhanqi: use new api

### PR DESCRIPTION
**Pull request by steven7851**
With the new api does not need to download the entire page.
Delete useless code "stream_weight", and make http as live for best quality.

### CODE REVIEW

**Comment by gravyboat** [comment]
Can we get some tests to go along with this? These are pretty big changes to not have any.

File: src/streamlink/plugins/zhanqi.py [hunk 1]
```diff
@@ -62,25 +30,17 @@ class Zhanqitv(Plugin):
...
-
-        res = http.get(API_URL.format(roomID=roomID))
+        res = http.get(API_URL.format(channel))
         room = http.json(res, schema=_room_schema)
         if not room:
             self.logger.info("Not a valid room url.")
             return
 
-        if room["status"] != STATUS_ONLINE:
+        if room["status"] != 4:
```
Comment (beardypig) [inline src/streamlink/plugins/zhanqi.py]: Why remove the constant? Seems to me that having the code read `room["status"] != STATUS_ONLINE` is more understandable than `room["status"] != 4`.

**Comment by gravyboat** [comment]
@beardypig Are we good to merge this?

**Comment by gravyboat** [comment]
Unless there are further concerns I would like to get this merged today.

**Comment by beardypig** [comment]
It looks OK. I haven't tested it though.

**Comment by gravyboat** [comment]
I haven't either. I'm going to merge it since I don't feel like doing the work to set it up so I can connect to a Chinese stream. Plus I trust that @steven7851 wouldn't open a PR that doesn't work.

**Comment by gravyboat** [comment]
Thanks @steven7851!

**Pull request merged**